pkg/watcher/results/results.go

```diff
@@ -61,10 +61,7 @@ func (c *Client) Put(ctx context.Context, o metav1.Object, opts ...grpc.CallOpti
 // ensureResult gets the Result corresponding to the Object, or creates a new
 // one.
 func (c *Client) ensureResult(ctx context.Context, o metav1.Object, opts ...grpc.CallOption) (*pb.Result, error) {
-	name, ok := o.GetAnnotations()[annotation.Result]
-	if !ok {
-		name = result.FormatName(o.GetNamespace(), c.defaultName(o))
-	}
+	name := resultName(o)
```

Comment:
Have a question here: if a taskRun and a PipelineRun in the same namespace shares the same name, and they have nothing to do with each other, then their result name here will be the same, will this cause a problem? Like they will be put under the same result by mistake?